hail/src/main/scala/is/hail/sparkextras/RepartitionedOrderedRDD2.scala

```diff
@@ -55,20 +55,74 @@ class RepartitionedOrderedRDD2 private (@transient val prev: RVD, @transient val
     val ordPartition = partition.asInstanceOf[RepartitionedOrderedRDD2Partition]
     val pord = kOrd.intervalEndpointOrdering
     val range = ordPartition.range
-    val ur = new UnsafeRow(typ.rowType)
-    val key = new SelectFieldsRow(ur, typ.kFieldIdx)
-
-    Iterator.single { (ctx: RVDContext) =>
-      ordPartition.parents.iterator
-        .flatMap { parentPartition =>
-          prevCRDD.iterator(parentPartition, context).flatMap(_(ctx))
-        }.dropWhile { ptr =>
-          ur.set(ctx.r, ptr)
-          pord.lt(key, range.left)
-        }.takeWhile { ptr =>
-          ur.set(ctx.r, ptr)
-          pord.lteq(key, range.right)
+
+    Iterator.single { (outerCtx: RVDContext) =>
+      new Iterator[Long] {
+        private[this] val innerCtx = outerCtx.freshContext()
+        private[this] val outerRegion = outerCtx.region
+        private[this] val innerRegion = innerCtx.region
+        private[this] val parentIterator = ordPartition.parents.iterator.flatMap(p => prevCRDD.iterator(p, context).next().apply(innerCtx))
```

Comment:
You changed from `flatMap` to `next`. I'm not sure we can assume a ContextRDD has exactly one element per partition.

Reply:
really? oh, I'll change that.